src/main/java/org/wildfly/security/audit/SyslogAuditEndpoint.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.audit;
+
+import static org.wildfly.common.Assert.checkNotNullParam;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.logging.Level;
+
+import org.jboss.logmanager.ExtLogRecord;
+import org.jboss.logmanager.handlers.SyslogHandler;
+import org.jboss.logmanager.handlers.SyslogHandler.Facility;
+import org.jboss.logmanager.handlers.SyslogHandler.Protocol;
+
+/**
+ * An {@link AuditEndpoint} that logs to syslog.
+ *
+ * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
+ */
+public class SyslogAuditEndpoint implements AuditEndpoint {
+
+    private volatile boolean accepting = true;
+
+    private final SyslogHandler syslogHandler;
+
+    /**
+     *
+     */
+    SyslogAuditEndpoint(Builder builder) throws IOException {
+        syslogHandler = new SyslogHandler(checkNotNullParam("serverAddress", builder.serverAddress), builder.port, Facility.SECURITY,
+                null, builder.tcp ? Protocol.TCP : Protocol.UDP, checkNotNullParam("hostName", builder.hostName));
+
+    }
+
+    @Override
+    public void accept(EventPriority t, String u) throws IOException {
+        if (!accepting) return;
+
+        synchronized(this) {
+            if (!accepting) return;
+
+            syslogHandler.doPublish(new ExtLogRecord(toLevel(t), u, SyslogAuditEndpoint.class.getName()));
```

Comment:
If we want the actual class/method that does the audit checking to show up in the log, this method will have to accept the FQCN of the first Elytron class handling the audit event (e.g. SecurityIdentity.class.getName()) which allows these fields to populate correctly.